python/pycolfits.py

```diff
@@ -13,10 +13,19 @@ def readfrom(filename, lower_case=False, upper_case=False, **kwargs):
     hdr = hdulist[1].header
     data = hdulist[1].data[0]
     colnames = hdulist[1].data.dtype.names
+    
+    ## check the length of columns to avoid any empty ones
+    checkcols = []
+    for cname in colnames:
+        try:
+            x = len(data[cname])
+            checkcols.append(cname)
+        except:
+            pass
```

Comment:
```suggestion
    checkcols = [x.name for x in hdulist[1].columns if re.search(r'[(,]0[),]', x.dim) is None]
```
A more elegant approach, scanning the dimensions definitions for zeros. Needs adding `import re`.